src/mmw/js/src/modeling/models.js

```diff
@@ -1468,6 +1468,124 @@ var ScenarioModel = Backbone.Model.extend({
         this.set('modification_hash', hash);
     },
 
+    /**
+     * Given an aggregate of GWLF-E modifications, validates them
+     * and warns about errors.
+     *
+     * In the future we may incorporate these warnings into a UI of
+     * some sort. For now, they are simply logged to the console.
+     */
+    validateGwlfeModifications: function(mods) {
+        var get = function(key) {
+            if (mods.hasOwnProperty(key)) {
+                return mods[key];
+            } else {
+                return 0;
+            }
+        };
+
+        if (get('n26') > 100) {
+            console.warn(
+                'GWLF-E: Crop Tillage Practice treated land use' +
+                'is too high: ' + round(get('n26'), 2) + '%'
+            );
+        }
+
+        if (get('n25') + get('n26') + get('n28b') > 100) {
+            console.warn(
+                'GWLF-E: Total treated percent of Cropland too high: ' +
+                round(get('n25') + get('n26') + get('n28b'), 2) + '%'
+            );
+        }
+
+        return true;
+    },
+
+    /**
+     * Converts current modifications into a single object with key / value
+     * pairs for overriding the baseline data model. Useful for aggregating
+     * the final value when multiple BMPs target the same value in a GMS file.
+     */
+    aggregateGwlfeModifications: function() {
+        var gisData = App.currentProject.get('gis_data'),
+            overrides = this.get('modifications').pluck('output'),
+            input = {
+                'CN__1': [], // Cropland Curve Number
+                'n26': [],   // Crop Tillage Practice Application Percentage
+                'n65': [],   // Crop Tillage Practice Nitrogen Efficiency
+                'n73': [],   // Crop Tillage Practice Phosphorus Efficiency
+                'n81': [],   // Crop Tillage Practice Sediments Efficiency
+            },
+            output = {};
+
+        // For every override, if its key is present in `input`, add its
+        // value to the array to be aggregated later. Else, just add it
+        // to `output` directly.
+        _.forEach(overrides, function(o) {
+            _.forEach(o, function(value, key) {
+                if (['CN__1', 'n26'].indexOf(key) >= 0) {
```

Comment:
In theory I think many of these keys should be set up as named constants; in practice it may not be worth undertaking that change now.